error: passing ‘const MyDiagonalTen’ as ‘this’ argument discards qualifiers

I'm having trouble getting the following example to work (although the first two examples in the README work fine for me). Very new to C++ so could be a bug or I could be missing something obvious.

```cpp
#include <Eigen/Core>
#include <Spectra/SymEigsSolver.h>
#include <iostream>

using namespace Spectra;

// M = diag(1, 2, ..., 10)
class MyDiagonalTen
{
public:
    using Scalar = double;  // A typedef named "Scalar" is required
    int rows() { return 10; }
    int cols() { return 10; }
    // y_out = M * x_in
    void perform_op(const double *x_in, double *y_out) const
    {
        for (int i = 0; i < rows(); i++)
        {
            y_out[i] = x_in[i] * (i + 1);
        }
    }
};

int main()
{
    MyDiagonalTen op;
    SymEigsSolver<MyDiagonalTen> eigs(op, 3, 6);
    eigs.init();
    eigs.compute(SortRule::LargestAlge);
    if (eigs.info() == CompInfo::Successful)
    {
        Eigen::VectorXd evalues = eigs.eigenvalues();
        std::cout << "Eigenvalues found:\n"
                  << evalues << std::endl;
    }

    return 0;
}
```

I see:

```
g++ tmp.cpp -o tmp && "/dabox/hayes/spectra/examples/"tmp
tmp.cpp: In member function ‘void MyDiagonalTen::perform_op(const double*, double*) const’:
tmp.cpp:17:34: error: passing ‘const MyDiagonalTen’ as ‘this’ argument discards qualifiers [-fpermissive]
   17 |         for (int i = 0; i < rows(); i++)
      |                                  ^
tmp.cpp:12:9: note:   in call to ‘int MyDiagonalTen::rows()’
   12 |     int rows() { return 10; }
      |         ^~~~
In file included from /usr/local/include/Spectra/SymEigsBase.h:22,
                 from /usr/local/include/Spectra/SymEigsSolver.h:12,
                 from tmp.cpp:2:
/usr/local/include/Spectra/MatOp/internal/ArnoldiOp.h: In instantiation of ‘Spectra::ArnoldiOp<Scalar, OpType, Spectra::IdentityBOp>::Index Spectra::ArnoldiOp<Scalar, OpType, Spectra::IdentityBOp>::rows() const [with Scalar = double; OpType = MyDiagonalTen; Spectra::ArnoldiOp<Scalar, OpType, Spectra::IdentityBOp>::Index = long int]’:
/usr/local/include/Spectra/LinAlg/Arnoldi.h:119:59:   required from ‘Spectra::Arnoldi<Scalar, ArnoldiOpType>::Arnoldi(ArnoldiOpType&&, Spectra::Arnoldi<Scalar, ArnoldiOpType>::Index) [with Scalar = double; ArnoldiOpType = Spectra::ArnoldiOp<double, MyDiagonalTen, Spectra::IdentityBOp>; Spectra::Arnoldi<Scalar, ArnoldiOpType>::Index = long int]’
/usr/local/include/Spectra/LinAlg/Lanczos.h:52:62:   required from ‘Spectra::Lanczos<Scalar, ArnoldiOpType>::Lanczos(T&&, Spectra::Lanczos<Scalar, ArnoldiOpType>::Index) [with T = Spectra::ArnoldiOp<double, MyDiagonalTen, Spectra::IdentityBOp>; Scalar = double; ArnoldiOpType = Spectra::ArnoldiOp<double, MyDiagonalTen, Spectra::IdentityBOp>; Spectra::Lanczos<Scalar, ArnoldiOpType>::Index = long int]’
/usr/local/include/Spectra/SymEigsBase.h:237:37:   required from ‘Spectra::SymEigsBase<OpType, BOpType>::SymEigsBase(OpType&, const BOpType&, Spectra::SymEigsBase<OpType, BOpType>::Index, Spectra::SymEigsBase<OpType, BOpType>::Index) [with OpType = MyDiagonalTen; BOpType = Spectra::IdentityBOp; Spectra::SymEigsBase<OpType, BOpType>::Index = long int]’
/usr/local/include/Spectra/SymEigsSolver.h:158:69:   required from ‘Spectra::SymEigsSolver<OpType>::SymEigsSolver(OpType&, Spectra::SymEigsSolver<OpType>::Index, Spectra::SymEigsSolver<OpType>::Index) [with OpType = MyDiagonalTen; Spectra::SymEigsSolver<OpType>::Index = long int]’
tmp.cpp:27:47:   required from here
/usr/local/include/Spectra/MatOp/internal/ArnoldiOp.h:120:50: error: passing ‘const MyDiagonalTen’ as ‘this’ argument discards qualifiers [-fpermissive]
  120 |     inline Index rows() const { return m_op.rows(); }
      |                                                  ^
tmp.cpp:12:9: note:   in call to ‘int MyDiagonalTen::rows()’
   12 |     int rows() { return 10; }
      |         ^~~~

[Done] exited with code=1 in 2.437 seconds
```

--- a/README.md
+++ b/README.md
@@ -2,8 +2,6 @@
 
 [![Build Status](https://travis-ci.org/yixuan/spectra.svg?branch=master)](https://travis-ci.org/yixuan/spectra) ![Basic CI](https://github.com/yixuan/spectra/workflows/Basic%20CI/badge.svg) [![codecov](https://codecov.io/gh/yixuan/spectra/branch/master/graph/badge.svg)](https://codecov.io/gh/yixuan/spectra)
 
-
-
 > **NOTE**: Spectra 1.0.0 is released, with a lot of
 > API-breaking changes. Please see the [migration guide](MIGRATION.md)
 > for a smooth transition to the new version.
@@ -69,6 +67,7 @@ of this object can then be called to conduct the computation and retrieve the
 eigenvalues and/or eigenvectors.
 
 Below is a list of the available eigen solvers in **Spectra**:
+
 - [SymEigsSolver](https://spectralib.org/doc/classSpectra_1_1SymEigsSolver.html):
 For real symmetric matrices
 - [GenEigsSolver](https://spectralib.org/doc/classSpectra_1_1GenEigsSolver.html):
@@ -191,8 +190,8 @@ class MyDiagonalTen
 {
 public:
     using Scalar = double;  // A typedef named "Scalar" is required
-    int rows() { return 10; }
-    int cols() { return 10; }
+    int rows() const { return 10; }
+    int cols() const { return 10; }
     // y_out = M * x_in
     void perform_op(const double *x_in, double *y_out) const
     {

